test/Kestrel.Core.Tests/Http2ConnectionTests.cs

```diff
@@ -1932,8 +1928,6 @@ public Task HEADERS_Received_HeaderBlockContainsTEHeader_ValueIsNotTrailers_Stre
 
             // No END_STREAM HEADERS or DATA frame is received since the stream was aborted
             await WaitForConnectionStopAsync(expectedLastStreamId: 1, ignoreNonGoAwayFrames: false);
-
-            // TODO: Check logs
```

Comment:
I noticed these tests call `SendGoAwayAsync` too. Prior to these being added, only the `GOAWAY_` tests called this helper. Any other test that wanted to close the stream just closed the pipe. I think we should keep this consistent.

Reply:
GO_AWAY is relevant to the RST tests since it calls the Abort code path.

Reply:
I don't buy it. If it was relevant, these tests wouldn't still pass when removing the SendGoAwayAsync() call. 

The ProcessGoAwayFrameAsync() just cancels the pending read on the Http2Connection's Input PipeReader stopping the ProcessRequestsAsync() loop. Completing the PipeWriter on the other side via StopConnectionAsync() also stops the loop and aborts all the streams.